lib/cli-engine.js

```diff
@@ -506,6 +510,33 @@ CLIEngine.prototype = {
             ignore: ignoredPathsList
         };
 
+
+        /**
+         * Calculates the hash of the config file used to validate a given file
+         * @param  {string} filename The path of the file to retrieve a config object for to calculate the hash
+         * @returns {string}         the hash of the config
+         */
+        function hashOfConfigFor(filename) {
+            var config = configHelper.getConfig(filename);
```

Comment:
I might be wrong, but it looks like we'll be calling `getConfig(filename)` two times for each filename, once here and once in `executeOnFile()`. This is a pretty expensive operation, so it would be good to do it only once.